flink-connectors/flink-connector-cassandra/pom.xml

```diff
@@ -83,6 +85,10 @@ under the License.
 										<exclude>com.google.inject.**</exclude>
 									</excludes>
 								</relocation>
+								<relocation>
+									<pattern>io.netty</pattern>
+									<shadedPattern>org.apache.flink.shaded.netty4.io.netty</shadedPattern>
```

Comment:
1) We haven't written them down anywhere; but there are examples on how the pattern should look like in many modules. (Granted, there are some relocations that don't follow the points I made; standardizing them is a work-in-progress) I would generally refer to existing relocations instead of some document that is bound to be out-dated at some point.
2) Currently we shade&relocate dependencies that already caused conflicts.
3) Yes, but there is no better solution. While it is possible to re-use some dependencies from Flink this tends to be a maintaining nightmare, as an upgrade to a dependency can suddenly cause issues in another part. The netty shading is a perfect example of this: we replaced the netty dependency and now the connector is broken.